app/src/lib/stores/helpers/branch-pruner.ts

```diff
@@ -0,0 +1,153 @@
+    if (this.timer !== null) {
+      fatalError(
+        `A background prune task is already active and cannot begin pruning on ${
+          this.repository.name
+        }`
+      )
+    }
+
+    await this.pruneLocalBranches()
+    this.timer = window.setInterval(
+      () => this.pruneLocalBranches(),
+      BackgroundPruneMinimumInterval
+    )
+  }
+
+  public stop() {
+    if (this.timer === null) {
+      return
+    }
+
+    clearInterval(this.timer)
+    this.timer = null
+  }
+
+  private async findBranchesMergedIntoDefaultBranch(
+    repository: Repository,
+    defaultBranch: Branch
+  ): Promise<ReadonlyArray<string> | null> {
+    const gitStore = this.gitStoreCache.get(repository)
+    return (
+      (await gitStore.performFailableOperation(() =>
+        getMergedBranches(repository, defaultBranch.name)
+      )) || null
+    )
+  }
+
+  private async pruneLocalBranches(): Promise<void> {
+    if (this.repository.gitHubRepository === null) {
+      return
+    }
+
+    // Get the last time this repo was pruned
+    const lastPruneDate = await this.repositoriesStore.getLastPruneDate(
+      this.repository
+    )
+
+    // Only prune if it's been at least 24 hours since the last time
+    const dateNow = moment()
+    const threshold = dateNow.subtract(24, 'hours')
+
+    if (lastPruneDate !== null && threshold.isBefore(lastPruneDate)) {
+      log.info(
+        `Last prune took place ${moment(lastPruneDate).from(
+          dateNow
+        )} - skipping`
+      )
+      return
+    }
+
+    // Get list of branches that have been merged
+    const { branchesState } = this.repositoriesStateCache.get(this.repository)
+    const { defaultBranch } = branchesState
+
+    if (defaultBranch === null) {
+      return
+    }
+
+    const mergedBranches = await this.findBranchesMergedIntoDefaultBranch(
+      this.repository,
+      defaultBranch
+    )
+
+    if (mergedBranches === null) {
+      log.info('No branches to prune.')
+      return
+    }
+
+    // Get all branches that exist on remote
+    const localBranches = branchesState.allBranches.filter(
+      x => x.type === BranchType.Local
+    )
+
+    // Create array of branches that can be pruned
+    const branchesReadyForPruning = new Array<Branch>()
+    for (const branch of mergedBranches) {
+      const localBranch = localBranches.find(
+        localBranch =>
+          localBranch.remote !== null && localBranch.name === branch
+      )
+
+      if (
+        localBranch !== undefined &&
+        !ReservedBranches.includes(branch.toLocaleLowerCase())
+      ) {
+        branchesReadyForPruning.push(localBranch)
+      }
+    }
+
+    log.info(
+      `Pruning ${branchesReadyForPruning.length} refs from '${
+        this.repository.name
+      } using '${defaultBranch.name} (${defaultBranch.tip.sha})' as base branch`
+    )
+
+    const gitStore = this.gitStoreCache.get(this.repository)
+    await gitStore.performFailableOperation(async () => {
```

Comment:
Oh, I see you've got the outer `performFailableOperation` here as well as the one I suggested.

I don't think we need that duplication - if the inner wrapper encounters an error it'll proceed and try to delete the next branch in the list (the outer one would give up on the first failure and you'd have to wait 24 hours for it to go again).